src/lock.rs

```diff
@@ -64,24 +64,8 @@ impl FromStr for DigestAlgorithm {
     }
 }
 
-impl Ord for DigestAlgorithm {
-    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
-        match self {
-            DigestAlgorithm::SHA256 => match other {
-                DigestAlgorithm::SHA256 => std::cmp::Ordering::Equal,
-            },
-        }
-    }
-}
-
-impl PartialOrd for DigestAlgorithm {
-    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
-        Some(self.cmp(other))
-    }
-}
-
```

Comment:
This was explicitly implemented on purpose, as a way to force future algorithm additions to have a meaningful ordering defined between them (e.g. SHA312 > SHA256). I think there was a comment on the PR where either Jon or James requested it.

Reply:
Would that be in #38? I looked around but was not able to find it.

My opinion is that this has to either be implemented using `derive(PartialOrd, Ord)` (which is still the recommended route when a meaningful ordering is wanted, as it will use the order the variants are defined in) or there has to be a comment explaining why it is necessary to hand-implement this (ideally with a link to the discussion in the PR). But it should not be left like this, with a hand-rolled implementation and no explanation.

Otherwise, the next person doing infraduction will refactor this again! :D

Reply:
Yes, I agree. #38 was a big and complex PR so this detail slipped through the cracks. I think for now a comment in both places should be plenty (we can worry about manual implementations once we actually need them).

Reply:
Sounds good to me 👍